repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -135,13 +139,26 @@ def process_events(events):
         api.produce(RepositoriesSetupTasks(to_enable=list(to_enable_repos)))
 
 
+def verify_pes_data(path):
+    """Verify if PES data is available if not inhibit upgrade"""
+    if path is None or not os.path.isfile(path):
+        title = 'PES data is invalid (/etc/leapp/files/pes-data.json)'
+        summary = 'Read documentation at: https://access.redhat.com/articles/3664871 for more information ' \
+            'about how to retrieve the files'
+        reporting.report_generic(title=title, summary=summary, severity='high', flags=['inhibitor'])
+        raise StopActorExecutionError(message=title, details={'details': summary})
```

Comment:
Up to you, but I would raise this StopActorExecutionError either here or after the "try/except" at "scan_events". Not in both. Maybe you can call "verify_pes_data" before "parse_file" inside "scan_events" and if a problem is found raise an exception inside "verify_pes_data" that will be catch inside "scan_events".